src/fsharp/fsi/fsi.fs

```diff
@@ -1273,13 +1275,48 @@ type internal FsiDynamicCompiler
         resolutions,
         { istate with tcState = tcState.NextStateAfterIncrementalFragment(tcEnv); optEnv = optEnv }
 
+
+    member __.EvalDependencyManagerTextFragment (packageManager:DependencyManagerIntegration.IDependencyManagerProvider,m,path: string) =
+        let path = DependencyManagerIntegration.removeDependencyManagerKey packageManager.Key path
+        
+        match tcConfigB.packageManagerLines |> Map.tryFind packageManager.Key with
+        | Some lines -> tcConfigB.packageManagerLines <- Map.add packageManager.Key (lines @ [path,m]) tcConfigB.packageManagerLines
+        | _ -> tcConfigB.packageManagerLines <- Map.add packageManager.Key [path,m] tcConfigB.packageManagerLines
+
+        needsPackageResolution <- true
+         
+    member fsiDynamicCompiler.CommitDependencyManagerText (ctok, istate: FsiDynamicCompilerState, lexResourceManager, errorLogger) = 
+        if not needsPackageResolution then istate else
+        needsPackageResolution <- false
+ 
+        tcConfigB.packageManagerLines |> Seq.fold(fun istate kv ->
+            let packageManagerKey,packageManagerLines = kv.Key, kv.Value
+            match packageManagerLines with
+            | [] -> istate
+            | (_, m)::_ ->
+                let packageManagerTextLines = packageManagerLines |> List.map fst
+                match DependencyManagerIntegration.tryFindDependencyManagerByKey m packageManagerKey with
+                | None ->
+                    errorR(DependencyManagerIntegration.createPackageManagerUnknownError packageManagerKey m)
+                    istate
+                | Some packageManager ->
+                    match DependencyManagerIntegration.resolve packageManager tcConfigB.implicitIncludeDir "stdin.fsx" "stdin.fsx" m packageManagerTextLines with
+                    | None -> istate // error already reported
+                    | Some (loadScript, additionalIncludeFolders) -> 
+                        for folder in additionalIncludeFolders do 
+                            tcConfigB.AddIncludePath(m, folder, "")
```

Comment:
Why do we actually need include folders when we have the load-Script?

Reply:
@matthid AFAIR they affect the resolution of subsequently loaded .fsx as well as references.

Reply:
But what is the difference to the script containing `#I`?

Reply:
I suspect the choice to return both the script location and additional include folders was taken in context of ScriptCs integration but don't remember for sure.

I'd need to check where that second value is used in context of FSI.

Other advantages (beside troubleshooting) of having a temporary script loaded is caching (based on a hash of the package manager directive contents), and making it easier to load a graph of things using relative path from the top level temporary script.

Reply:
Note that I'm not against a temporary script at all. I'm just trying to follow the reasoning, because the spec doesn't really say or list pros/cons or something like that. From my perspective it just makes things more complicated. Also I don't really understand the caching argument because the package manager can still cache a list of paths and so can the compiler?

Reply:
There is still room to adjust the PackageManager integration signatures, and actually document it better in the code (I see there are no comments about the return value).

One concern though was that the return value has to be plain .net types rather than F# specific or relying on extra assembly, to allow easier interop via reflection from reflection / tooling perspective.

Are you concerned the temporary script is going to be a concern?